### ISSUE TITLE
Why no tutorial on how to deploy piranha.blog.razor?

### ISSUE THREAD

**Issue by LeadMagnet** [PiranhaCMS/piranha.core.docs#23]
A very brief section on using "dotnet new". Never seen it before, but OK. Always just "create new solution" but hey, I like learning things and stuff.

An even briefer subject on using SQL Server. Implementing with software I've never seen, yeah sure, lets see if this works.

Authentication using said SQL database, well I just cut-n-paste the one sentence on this topic.

So lets deploy...OK...how do I do that? I use IIS in production, so publish to IIS folder. Nope, not working.

There is zero documentation on deploying this software, in any configuration, on any platform.

Think one of you guys could find the time to write 2 documents. One for .Net Core stand-alone deployment. One for IIS deployment. Either one would be great.

Thanks,

Have A Nice Day!

**Comment by tidyui** [comment]
Hi there! The reason we haven't written anything on deployment is that there's nothing special about Piranha compared to a regular .NET Core application. In other words, whatever we write we're pretty sure that the official Microsoft documentation on deploying .NET Core will be more extensive and up-to-date than ours :)

Regarding the other topics you mention, creating a new project, setting up a database, authentication and so on, please elaborate on what you'd like us to add to the documentation. Writing good documentation is usually harder than writing the code!

Best regards

**Comment by LeadMagnet** [comment]
Yeah I feel your pain! Documentation is my nemesis, but ya gotta do it! 

The "official Microsoft documentation on deploying .NET Core" for Piranha hasn't worked. Admittedly, I know absolutely nothing about deploying a Core website. IIS, been doing that for decades.

As far as the database and authentication issues. I think the problem is I ran the solution "as is" in VS. WHich worked, but when I changed to SQL, nothing did. The current documentation for SQL setup (in my opionion) is unclear, blurred by the other choices. Same for authentication, unlcear as to what needs changing in the default nuget load. It ran fine in SQLLite, but I don't deploy SQLLite. I know everything you can on Webforms and MVC deployment, but Core is kicking the proverbial behind. So gonna stop.

Thanks for the reply, appreciate it. I've spent days on this and need a solution 2 days ago. 

All I can really recommend is approach a "how to" from the point of view that the reader has never seen a computer, much less .Net Core

:D

**Comment by tidyui** [comment]
@filipjansson can you provide some info on how to deploy on IIS, I know you’ve done this on your local computer!

**Comment by LeadMagnet** [comment]
The only "deployment" I've done is clicking the "Run" (F5) button in Visual Studio Professional 2019. Following your [installation instructions](http://piranhacms.org/docs/basics/project-templates) this ran just fine.

Getting the same project to run on local IIS has failed, for me anyway. Attempts to get it to run on Server 2016 and 2019 using IIS all failed. 

The (IIS reported) error is "web.config syntax error..line 0", but Core web.config files are built on the fly when publishing, I have no clue what I'm looking at for a Core web.config file so no idea what is wrong, especially since line zero is not readable, line 1 is the first line. 

After the changes to use SQL and SQL for authentication, it wouldn't run at all, even using "dotnet piranha.dll" instead of IIS. If there was error messages (command window, event viewer), I couldn't see them.

I'm sure this is down to my total lack of knowledge with Core, but right now I have no time, I need an "out of the box, just works" on IIS and SQL. That's the infrastructure I have in production.

Thanks

**Comment by filipjansson** [comment]
Hi!

I was trying to setup IIS to run asp.net core, but I´m not sure I managed to get it to work correctly.
Anyway, I was following the [official docs from Microsoft](https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/iis/?view=aspnetcore-3.1) for running asp.net core on IIS which explains a lot and how to install the module and so on.

BR
Filip

**Comment by LeadMagnet** [comment]
> Hi!
> 
> I was trying to setup IIS to run asp.net core, but I´m not sure I managed to get it to work correctly.
> Anyway, I was following the [official docs from Microsoft](https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/iis/?view=aspnetcore-3.1) for running asp.net core on IIS which explains a lot and how to install the module and so on.
> 
> BR
> Filip

I ran through that document so many times, made so many changes, I couldn't honestly say if it did any good.

In the end I used the older ASP.Net community blog project in .Net 2, which compiled and deployed in minutes on our infrastructure. Now that the customer has his blog, I can revisit this issue.

Turns out I needed to install both the version 2 and version 3.1 of the Web Hosting framework. That cleared the web.config error and got the main page to display.

So I clicked the "seed" button....and now cannot log in to the site using "admin / password". The seed procedure is not inserting data into the users table when configured for Identity and SQL Server. Those tables are all blank. The site works (I think) and displays the seed infor, navigation seems to work, going to "/manager" works but there are no credentials to log in inside the database.

So I'm getting somewhere....ever so slowly! :D I'll open a new ticket for this new issue.
